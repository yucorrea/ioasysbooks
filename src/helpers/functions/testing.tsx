import React from 'react';
import renderer from 'react-test-renderer';
import { render, RenderOptions } from '@testing-library/react-native';
import { ThemeProvider } from 'styled-components/native';

import theme from '../../config/theme';

export const matchSnapshotWithTheme = (component :  React.ReactElement) => {
  const tree = renderer.create(<ThemeProvider theme={theme}>{component}</ThemeProvider>).toJSON();
  expect(tree).toMatchSnapshot();
};

export const renderWithTheme = (component : React.ReactElement, options?: RenderOptions) => {
  return render(<ThemeProvider theme={theme}> {component} </ThemeProvider>, options);
};
