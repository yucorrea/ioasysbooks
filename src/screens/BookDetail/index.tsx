import React from 'react';
import { useNavigation, useRoute } from '@react-navigation/core';
import { ScrollView } from 'react-native';
import styled from 'styled-components/native';

import mark from '../../assets/mark.png';
import { IconButton } from '../../components/IconButton';

interface Book {
  imageUrl: string;
  title: string;
  pageCount: number;
  publisher: string;
  published: number;
  isbn10: string;
  isbn13: string;
  category: string;
  authors: Array<string>;
  language: string;
  description: string;
}

interface BookParams {
 book: Book;
}

export function BookDetail() {
  const { goBack } = useNavigation();
  const route = useRoute();

  const { book } = route.params as BookParams;
  return (
    <StyledContainer>
      <StyledScreen>
      <IconButton icon="arrow-back-outline" onPress={() => goBack()} />
        <StyledWrapper>
          <StyledBook source={{ uri: book.imageUrl }} resizeMode="contain" />
          <StyledTitle numberOfLines={2}>{book.title}</StyledTitle>
          <StyledAuthorsContainer>
            <StyledAuthor>{book.authors.toString()}</StyledAuthor>
          </StyledAuthorsContainer>

          <StyledInformationContainer>
            <StyledInformationTitle>Informações</StyledInformationTitle>
            <StyledDetail>
              <StyledLeft>Páginas</StyledLeft>
              <StyledRight>{book.pageCount} páginas</StyledRight>
            </StyledDetail>

            <StyledDetail>
              <StyledLeft>Editora</StyledLeft>
              <StyledRight>{book.publisher}</StyledRight>
            </StyledDetail>

            <StyledDetail>
              <StyledLeft>Publicação</StyledLeft>
              <StyledRight>{book.published}</StyledRight>
            </StyledDetail>

            <StyledDetail>
              <StyledLeft>Idioma</StyledLeft>
              <StyledRight>{book.language}</StyledRight>
            </StyledDetail>

            <StyledDetail>
              <StyledLeft>Título Original</StyledLeft>
              <StyledRight>{book.title} </StyledRight>
            </StyledDetail>

            <StyledDetail>
              <StyledLeft>ISBN-10</StyledLeft>
              <StyledRight>{book.isbn10} </StyledRight>
            </StyledDetail>

            <StyledDetail>
              <StyledLeft>ISBN-13</StyledLeft>
              <StyledRight>{book.isbn13} </StyledRight>
            </StyledDetail>

            <StyledDetail>
              <StyledLeft>Categoria</StyledLeft>
              <StyledRight>{book.category}</StyledRight>
            </StyledDetail>
          </StyledInformationContainer>

          <StyledReviewContainer>
            <StyledInformationTitle>Resenha da Editora</StyledInformationTitle>

            <StyledAbsoluteWrapper>
              <StyledMark source={mark} />
              <StyledDescription> {book.description}</StyledDescription>
            </StyledAbsoluteWrapper>
          </StyledReviewContainer>
        </StyledWrapper>
      </StyledScreen>
    </StyledContainer>
  );
}

const StyledContainer = styled.SafeAreaView`
  flex: 1;
  background: ${({ theme }) => theme.colors.background};
`;
const StyledScreen = styled.ScrollView.attrs({
  showsVerticalScrollIndicator:false,
  contentContainerStyle: {
    paddingHorizontal: 16,
    paddingTop: 16
  }
})``;

const StyledBook = styled.Image`
  height: 351px;
  margin-top: 12px;
  margin-bottom: 24px; ;
`;

const StyledWrapper = styled.View`
  padding: 12px 40px 40px 40px;
`;

const StyledTitle = styled.Text`
  font-size: 28px;
  font-family: ${({ theme }) => theme.fonts.medium};
  color: ${({ theme }) => theme.colors.title};
`;

const StyledAuthorsContainer = styled.View`
  flex-direction: row;
`;

const StyledAuthor = styled.Text`
  font-size: 12px;
  font-family: ${({ theme }) => theme.fonts.regular};
  color: ${({ theme }) => theme.colors.primary_light};
`;

const StyledInformationContainer = styled.View`
  margin-top: 32px;
`;

const StyledInformationTitle = styled.Text`
  margin-bottom: 14px;
  font-size: 16px;
  text-transform: uppercase;
  font-family: ${({ theme }) => theme.fonts.medium};
  color: ${({ theme }) => theme.colors.title};
`;

const StyledDetail = styled.View`
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  width: 100%;
  margin: 4px 0px;
`;

const StyledLeft = styled.Text`
  font-size: 12px;
  font-family: ${({ theme }) => theme.fonts.medium};
  color: ${({ theme }) => theme.colors.title};
`;

const StyledRight = styled.Text`
  font-size: 14px;
  font-family: ${({ theme }) => theme.fonts.regular};
  color: ${({ theme }) => theme.colors.text};
`;

const StyledReviewContainer = styled.View`
  width: 100%;
  margin-top: 16px;
`;

const StyledAbsoluteWrapper = styled.View`
  width: 95%;
  flex-direction: row;
`;

const StyledDescription = styled.Text`
  align-items: baseline;
  text-align: justify;
  font-size: 14px;
  font-family: ${({ theme }) => theme.fonts.regular};
  color: ${({ theme }) => theme.colors.text};
`;

const StyledMark = styled.Image`
  margin-bottom: 4px;
`;
