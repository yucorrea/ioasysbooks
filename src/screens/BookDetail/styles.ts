import styled from 'styled-components/native';

export const Container = styled.SafeAreaView`
  flex: 1;
  padding: 44px 40px 12px 16px;
  background: #f6f5ef;
`;

export const Book = styled.Image`
  height: 240px;
  height: 351px;
  margin-top: 12px;
  margin-bottom: 24px; ;
`;

export const Wrapper = styled.View`
  padding: 12px 40px 40px 40px;
`;

export const Title = styled.Text`
  font-size: 28px;
  font-family: ${({ theme }) => theme.fonts.medium};
  color: ${({ theme }) => theme.colors.title};
`;

export const AuthorsContainer = styled.View`
  flex-direction: row;
`;

export const Author = styled.Text`
  font-size: 12px;
  font-family: ${({ theme }) => theme.fonts.regular};
  color: ${({ theme }) => theme.colors.primary_light};
`;

export const InformationContainer = styled.View`
  margin-top: 32px;
`;

export const InformationTitle = styled.Text`
  font-size: 16px;
  text-transform: uppercase;
  font-family: ${({ theme }) => theme.fonts.medium};
  color: ${({ theme }) => theme.colors.title};
  margin-bottom: 14px;
`;

export const Detail = styled.View`
  width: 100%;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 4px 0px;
`;

export const Left = styled.Text`
  font-size: 12px;
  font-family: ${({ theme }) => theme.fonts.medium};
  color: ${({ theme }) => theme.colors.title};
`;

export const Right = styled.Text`
  font-size: 14px;
  font-family: ${({ theme }) => theme.fonts.regular};
  color: ${({ theme }) => theme.colors.text};
`;

export const ReviewContainer = styled.View`
  width: 100%;
  margin-top: 16px;
`;

export const AbsoluteWrapper = styled.View`
  width: 95%;
  flex-direction: row;
`;

export const Description = styled.Text`
  font-size: 14px;
  font-family: ${({ theme }) => theme.fonts.regular};
  color: ${({ theme }) => theme.colors.text};
  align-items: baseline;
  text-align: justify;
`;

export const Mark = styled.Image`
  margin-bottom: 4px;
`;
