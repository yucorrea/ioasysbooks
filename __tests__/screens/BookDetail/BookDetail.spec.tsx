import React from 'react';
import { matchSnapshotWithTheme, renderWithTheme } from '../../../src/helpers/functions/testing';

import BookDetail from '../../../src/screens/BookDetail';

describe('BookDetail Screen', () => {
  test('matches snapshot', () => {
    matchSnapshotWithTheme(<BookDetail />);
  });

  test('renders correctly', () => {
    renderWithTheme(<BookDetail />);
  });
});
