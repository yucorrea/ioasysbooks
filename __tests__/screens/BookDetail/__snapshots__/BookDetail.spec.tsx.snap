// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookDetail Screen matches snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "#ffffff",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "paddingHorizontal": 16,
        "paddingTop": 16,
      }
    }
    showsVerticalScrollIndicator={false}
    style={
      Array [
        Object {},
      ]
    }
  >
    <View>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        nativeID="animatedComponent"
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "borderColor": "rgba(51, 51, 51, 0.2)",
            "borderRadius": 16,
            "borderStyle": "solid",
            "borderWidth": 1,
            "height": 32,
            "justifyContent": "center",
            "opacity": 1,
            "width": 32,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontSize": 16,
              },
              undefined,
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
      <View
        style={
          Array [
            Object {
              "paddingBottom": 40,
              "paddingLeft": 40,
              "paddingRight": 40,
              "paddingTop": 12,
            },
          ]
        }
      >
        <Image
          resizeMode="contain"
          source={
            Object {
              "uri": "",
            }
          }
          style={
            Array [
              Object {
                "height": 351,
                "marginBottom": 24,
                "marginTop": 12,
              },
            ]
          }
        />
        <Text
          numberOfLines={2}
          style={
            Array [
              Object {
                "color": "#333333",
                "fontFamily": "Heebo-Medium",
                "fontSize": 28,
              },
            ]
          }
        >
          
        </Text>
        <View
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#AB2680",
                  "fontFamily": "Heebo-Regular",
                  "fontSize": 12,
                },
              ]
            }
          >
            
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 32,
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Heebo-Medium",
                  "fontSize": 16,
                  "marginBottom": 14,
                  "textTransform": "uppercase",
                },
              ]
            }
          >
            Informações
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 4,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 4,
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Medium",
                    "fontSize": 12,
                  },
                ]
              }
            >
              Páginas
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#999999",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 14,
                  },
                ]
              }
            >
              1
               páginas
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 4,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 4,
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Medium",
                    "fontSize": 12,
                  },
                ]
              }
            >
              Editora
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#999999",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 14,
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 4,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 4,
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Medium",
                    "fontSize": 12,
                  },
                ]
              }
            >
              Publicação
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#999999",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 14,
                  },
                ]
              }
            >
              0
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 4,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 4,
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Medium",
                    "fontSize": 12,
                  },
                ]
              }
            >
              Idioma
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#999999",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 14,
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 4,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 4,
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Medium",
                    "fontSize": 12,
                  },
                ]
              }
            >
              Título Original
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#999999",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 14,
                  },
                ]
              }
            >
              
               
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 4,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 4,
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Medium",
                    "fontSize": 12,
                  },
                ]
              }
            >
              ISBN-10
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#999999",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 14,
                  },
                ]
              }
            >
              
               
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 4,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 4,
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Medium",
                    "fontSize": 12,
                  },
                ]
              }
            >
              ISBN-13
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#999999",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 14,
                  },
                ]
              }
            >
              
               
            </Text>
          </View>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "space-between",
                  "marginBottom": 4,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 4,
                  "width": "100%",
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#333333",
                    "fontFamily": "Heebo-Medium",
                    "fontSize": 12,
                  },
                ]
              }
            >
              Categoria
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#999999",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 14,
                  },
                ]
              }
            >
              
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              Object {
                "marginTop": 16,
                "width": "100%",
              },
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#333333",
                  "fontFamily": "Heebo-Medium",
                  "fontSize": 16,
                  "marginBottom": 14,
                  "textTransform": "uppercase",
                },
              ]
            }
          >
            Resenha da Editora
          </Text>
          <View
            style={
              Array [
                Object {
                  "flexDirection": "row",
                  "width": "95%",
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../src/assets/images/mark.png",
                }
              }
              style={
                Array [
                  Object {
                    "marginBottom": 4,
                  },
                ]
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "alignItems": "baseline",
                    "color": "#999999",
                    "fontFamily": "Heebo-Regular",
                    "fontSize": 14,
                    "textAlign": "justify",
                  },
                ]
              }
            >
               
              
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
